apps/namadillo/src/App/Transfer/SelectToken.tsx

```diff
@@ -0,0 +1,379 @@
+import { Chain } from "@chain-registry/types";
+import { Modal, Stack } from "@namada/components";
+import { ModalTransition } from "App/Common/ModalTransition";
+import { Search } from "App/Common/Search";
+import {
+  connectedWalletsAtom,
+  getAvailableChains,
+  getChainRegistryByChainName,
+  getNamadaChainRegistry,
+  namadaRegistryChainAssetsMapAtom,
+} from "atoms/integrations";
+import { tokenPricesFamily } from "atoms/prices/atoms";
+import clsx from "clsx";
+import { useWalletManager } from "hooks/useWalletManager";
+import { KeplrWalletManager } from "integrations/Keplr";
+import { useAtom, useAtomValue } from "jotai";
+import { useMemo, useState } from "react";
+import { IoClose } from "react-icons/io5";
+import { AssetWithAmountAndChain } from "types";
+import { AddressDropdown } from "./AddressDropdown";
+import { ChainBadge } from "./ChainBadge";
+import { isNamadaAddress } from "./common";
+
+type SelectTokenProps = {
+  setSourceAddress: (address: string) => void;
+  sourceAddress: string;
+  destinationAddress: string;
+  isOpen: boolean;
+  onClose: () => void;
+  onSelect:
+    | ((
+        selectedAsset: AssetWithAmountAndChain,
+        newSourceAddress?: string
+      ) => void)
+    | undefined;
+  keplrWalletManager?: KeplrWalletManager | undefined;
+  assetsWithAmounts: AssetWithAmountAndChain[];
+};
+
+export const SelectToken = ({
+  sourceAddress,
+  destinationAddress,
+  setSourceAddress,
+  isOpen,
+  onClose,
+  onSelect,
+  assetsWithAmounts,
+  keplrWalletManager,
+}: SelectTokenProps): JSX.Element | null => {
+  const [filter, setFilter] = useState("");
+  const [selectedNetwork, setSelectedNetwork] = useState<string | null>(null);
+  const [isConnectingKeplr, setIsConnectingKeplr] = useState(false);
+  const [connectedWallets, setConnectedWallets] = useAtom(connectedWalletsAtom);
+  const chainAssets = useAtomValue(namadaRegistryChainAssetsMapAtom);
+  const chainAssetsMap = Object.values(chainAssets.data ?? {});
+  const ibcChains = useMemo(getAvailableChains, []);
+
+  const allChains = [...ibcChains, getNamadaChainRegistry(false).chain];
```

Comment:
Is it ok to ignore housefire here?

Reply:
WDYT about adding new atom that returns all the chains? This way we would not call code from `functions` directly?